internal/cmd/gocdk/demo_test.go

```diff
@@ -43,14 +43,14 @@ func TestAddDemo(t *testing.T) {
 	}
 	ctx := context.Background()
 
-	if err := run(ctx, pctx, []string{"init", "-m", "test", "--allow-existing-dir", dir}, new(bool)); err != nil {
+	if err := init_(ctx, pctx, dir, "testmodpath", true); err != nil {
 		t.Fatalf("run init error: %+v", err)
 	}
 
 	// Call the main package run function as if 'add-demo' were being called
 	// from the command line for each of the demos.
 	for _, demo := range allDemos {
-		if err := run(ctx, pctx, []string{"demo", demo.name}, new(bool)); err != nil {
+		if err := run(ctx, pctx, []string{"demo", "add", demo.name}, new(bool)); err != nil {
```

Comment:
FYI, I used the pattern of calling the function directly in tests most of the time (e.g., `init_` above), but calling `run` for the function under test (e.g., `run` here). WDYT?

Reply:
Using `run` over time seems like it would be closer to how the user would experience it, which would make the tests less brittle over time. Can you help me understand the case for calling command functions directly in tests? I think I'd be more okay with using helper functions shared between tests and prod code that work on the parsed arguments.

Reply:
Mostly just that it's simpler -- you get to call a function with named arguments that probably match what you have in hand. With `run`, you have to convert those arguments to flags etc.; it's harder to write and harder to read.

I left the use of `run` for the thing actually being tested; e.g., `demo_test.go` uses `run` to call `gocdk demo add`, as I agree it's important to test as close to what the user experience as possible. But I don't think it's necessary to do that for things that aren't directly under test.

We can revisit this if you want to discuss more, it's pretty easy to change.

Reply:
Also I notice that I forgot to update this file to use a new test helper, `newTestProject`, which gets rid of the call to `init_` here anyway.